pkg/skaffold/build/kaniko/run.go

```diff
@@ -43,11 +45,53 @@ func runKaniko(ctx context.Context, out io.Writer, artifact *v1alpha3.Artifact,
 	dockerfilePath := artifact.DockerArtifact.DockerfilePath
 
 	initialTag := util.RandomID()
-	tarName := fmt.Sprintf("context-%s.tar.gz", initialTag)
-	if err := docker.UploadContextToGCS(ctx, artifact.Workspace, artifact.DockerArtifact, cfg.BuildContext.GCSBucket, tarName); err != nil {
-		return "", errors.Wrap(err, "uploading tar to gcs")
+	volumes := []v1.Volume{{
+		Name: constants.DefaultKanikoSecretName,
+		VolumeSource: v1.VolumeSource{
+			Secret: &v1.SecretVolumeSource{
+				SecretName: cfg.PullSecretName,
+			},
+		},
+	}}
+
+	volumeMounts := []v1.VolumeMount{{
+		Name:      constants.DefaultKanikoSecretName,
+		MountPath: "/secret",
+	}}
+
+	context := ""
+	if cfg.BuildContext.GCSBucket != "" {
```

Comment:
I would change this to a switch statement, and maybe break out the implementations into their own functions

Reply:
The function could be something like func(ctx, artifact, cfg.BuildContext, cfg.Namespace, initialTag) (context, func())